src/session_impl.cpp

```diff
@@ -5548,6 +5554,28 @@ namespace aux {
 			, boost::ref(m_alerts), _1, cb), salt);
 	}
 
+	void session_impl::dht_get_peers(sha1_hash const& info_hash)
+	{
+		if (!m_dht) return;
```

Comment:
I may have been a bit lazy when I wrote dht_put_item(). The idea is to (similar to all asio async_* operations) for every async operation that's initiated, there is exactly one response to it. Which may be a failure or an operation_aborted. Having such guarantee is a quite nice property and simplifies reasoning about them.